diesel/src/pg/connection/result.rs

```diff
@@ -167,6 +167,14 @@ impl DatabaseErrorInformation for PgErrorInformation {
     fn constraint_name(&self) -> Option<&str> {
         get_result_field(self.0.as_ptr(), ResultField::ConstraintName)
     }
+
+    fn statement_position(&self) -> Option<f32> {
+        if let Some(str_pos) = get_result_field(self.0.as_ptr(), ResultField::StatementPosition) {
+            Some(str_pos.parse::<f32>().expect("Invalid number"))
+        } else {
+            None
+        }
+    }
```

Comment:
We should not panic here, instead return `None` for error cases.

```suggestion 
       let str_pos = get_result_field(self.0.as_ptr(), ResultField::StatementPosition)?;
       str_pos.parse::<i32>().ok()
```